tutorials/holoscan_debugging/README.md

```diff
@@ -0,0 +1,302 @@
+
+Run the script to get started:
+```sh
+./tutorials/holoscan_debugging/debug_pdb.sh
+```
+
+The script runs through the following steps:
+1. Builds the HoloHub container environment based on the Holoscan SDK NGC container.
+2. Builds the Endoscopy Tool Tracking Python application in the container environment. By default we use the
+  `debug` build mode to generate detailed C++ debugging symbols for the Endoscopy Tool Tracking application.
+3. Launches the Endoscopy Tool Tracking application with `pdb`, which is invoked via the Python interpreter:
+```bash
+python3 -m pdb <command --args ...>
+```
+
+This command launches the Python version of the Endoscopy Tool Tracking application with a breakpoint set on the
+very first line. From this point you can set additional breakpoints, inspect application state, and control program
+execution.
+
+For instance, the following snippet sets a breakpoint and then inspects the value of `self.source.lower()` during
+pipeline setup in the app `compose` method:
+```py
+(Pdb) break endoscopy_tool_tracking.py:77
+Breakpoint 1 at /workspace/holohub/applications/endoscopy_tool_tracking/python/endoscopy_tool_tracking.py:77
+(Pdb) continue
+...
+> /workspace/holohub/applications/endoscopy_tool_tracking/python/endoscopy_tool_tracking.py(77)compose()
+-> if self.source.lower() == "aja":
+(Pdb) p self.source.lower()
+'replayer'
+(Pdb)
+...
+```
+
+You can also add a breakpoint directly in the `.py` source code before running a program by adding a `breakpoint()` statement.
+
+From here we recommend referring to [`pdb` documentation](https://docs.python.org/3/library/pdb.html) for common
+commands and debugging strategies.
+
+## Debugging Symbols for Legacy Holoscan SDK Versions
+
+In Holoscan SDK v2.3 we began packaging debugging symbols as part of the libraries distributed in
+[NGC Holoscan Containers](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/clara-holoscan/containers/holoscan), with the goal
+of improving ease of development and debugging. This change comes at a small cost to size and performance of the Holoscan SDK
+binary distribution. But what about older versions of Holoscan SDK containers?
+
+If you are using a legacy Holoscan SDK container (earlier than v2.3) for your development, your container does not come with
+debugging symbols pre-packaged. However, you can rebuild Holoscan SDK from source to enable interactive debugging.
+We provide utilities as part of Holoscan SDK open source code to help you generate debugging
+symbols that will allow you to use tools such as `gdb` and `pdb` with legacy Holoscan SDK.
+
+The following script provides the necessary steps to rebuild Holoscan SDK version with debugging symbols and then
+set up for debugging with `gdb`:
+
+```bash
+./tutorials/holoscan_debugging/debug_legacy.sh <build_type> <sdk_tag> <ngc_tag>
+```
+
+The script does the following:
+1. Builds the specified Holoscan SDK version with the specified build type in a temporary tutorial folder.
```

Comment:
For build type: do we prefer `rel-debug` than `debug`? When do we use `debug` rather than `rel-debug`? It would be helpful to provide guidance here.

Reply:
In my view that discussion is better left in the introduction, added a sentence linking to that explanatory section.